### ISSUE TITLE
expand rather than overwrite with a ExecStart= systemd drop-in

### ISSUE THREAD

**Issue by adrelanos** [systemd/systemd#21461]
**Is your feature request related to a problem? Please describe.**

A distribution package daemon is shipping a fine systemd unit. That daemon that gets started `--comes --with --some  --parameters`. I am OK with these parameters and would like to keep receiving updates as decided by the distribution package maintainer. However, I'd would like to add a parameter of my own.

Now the only solution I am aware of is this:

`/lib/systemd/system/daemon-name.service.d/30_drop-in.conf`

```
ExecStart=
ExecStart=/path/to/daemon --comes --with --some --parameters --my-extra-parameter
```

**Describe the solution you'd like**

`/lib/systemd/system/daemon-name.service.d/30_drop-in.conf`

```
ExecStart=%append% --my-extra-parameter
```

`%append%` is maybe non-idea. Perhaps someone has a better idea how a nice syntax would look like.

**Describe alternatives you've considered**

None.

----

This might also be useful for `ExecStartPre=`. Also other keywords such as

* `%param-prepend%` (resulting `ExecStart=/path/to/daemon --prepend --comes --with --some --parameters`),
* `%param-replace% pattern replacement`,
* or `%program-replace% /path/to/different-program` (resulting in `ExecStart=/path/to/different-program --prepend --comes --with --some --parameters`)
would be useful.

**Comment by OJFord** [comment]
What if the original unit has multiple `ExecStart`s, to which will your extra parameter be appended?

**Comment by adrelanos** [comment]
Good point. Unless someone has a great idea how to handle that case... How about...?

* Only 1 ExecStart + append: Do as above.
* Multiple ExecStart + append: Should be considered a syntax error and the systemd unit should fail.

**Comment by rektide** [comment]
Inability to compose drop ins is frustrating. This would be such a huge QoL boost for drop-ins.

**Comment by AlexDumanskiy** [comment]
Maybe a solution could be to allow multiple ExecStart entries?

Concatenate values much like other software does with cmdline arguments, trim before concatenation to account for all service types, except for oneshot?
Trim each ExecStart individually before concatenation to minimize surprises.

Docker:
`ENTRYPOINT [ "/bin/sh", "-c", "cowsay", "moo" ]`

Java, (ProcessBuilder):
`var pb = new ProcessBuilder(Arrays.asList("java", "-jar", "-Xmx1024m", "app.jar");`

**Comment by adrelanos** [comment]
I like the idea of `ExecStartArgs=`.

> What if the original unit has multiple `ExecStart`s, to which will your extra parameter be appended?

* `ExecStart=` considered same as `ExecStart[1]`
* The second `ExecStart=` is considered `ExecStart[2]`.
* And to inject additional arguments:
  * `ExecStart[1]` / `ExecStartArgs[1]=`.
  * `ExecStart[2]` / `ExecStartArgs[2]=`.
  * `ExecStart[3]` / `ExecStartArgs[3]=`.

`ExecStartArgs[1]=` etc. could even be declared several times by several drop-ins. In that case, keep appending.

**Comment by ElSamhaa** [comment]
How about a one and only simple rule; concatenate `ExecStart`s by the order the are set and reset when an empty `ExecStart=` is encountered.

**Comment by adrelanos** [comment]
ElSamhaa:
> How about a one and only simple rule; concatenate `ExecStart`s by the order the are set and reset when an empty `ExecStart=` is encountered.

It would be nice if systemd was designed today. But since multiple 
ExecStart= nowadays as already implemented means they all will be 
executed, this would be an API (system configuration language) change. I 
somehow guess this would be unlikely to be accepted.
